src/azul/service/manifest_service.py

```diff
@@ -441,7 +460,7 @@ def _use_cached_manifest(self, generator: 'ManifestGenerator', object_key: str)
                                object_key, self.file_name_tag)
         else:
             file_name = None
-        return file_name
+        return file_name, use_cached_manifest
```

Comment:
Consider eliminating the tuple return and instead use a None file name to signal that cached manifest is invalid or can't be used. When use_content_disposition_file_name is False this should return the last component of the object key. You'll need to call use_content_disposition_file_name again in the caller to determine whether to sign the URL with the CD or not.